.github/workflows/documentation.yaml

```diff
@@ -0,0 +1,109 @@
+# - https://squidfunk.github.io/mkdocs-material/publishing-your-site/?h=deploy#material-for-mkdocs
+
+name: mkdocs
+
+on:
+  push:
+    branches: main
+  pull_request:
+    branches: main
+  release:
+    types: [published]
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+jobs:
+  documentation:
+
+    runs-on: ubuntu-latest
+
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+
+    # Only restrict concurrency for non-PR jobs
+    concurrency:
+      group: pkgdown-${{ github.event_name != 'pull_request' || github.run_id }}
+
+    steps:
+
+       ##### Github actions and Python stuff #####
+
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+      - uses: actions/cache@v2
+        with:
+          key: ${{ github.ref }}
+          path: .cache
+      - run: pip install mkdocs-material
+
+
+        ##### Rust stuff #####
+
+      - run: echo "rust-toolchain=nightly" >>"$GITHUB_ENV"
+      - name: Set up Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.rust-toolchain }}
+      - name: Cache Rust
+        uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: ./src/rust/
+          cache-on-failure: "true"
+
+      - name: Install ubuntu SYSTEM REQUIREMENTS
+        run: |
+          sudo apt-get update \
+          && sudo apt-get install -y \
+            libfontconfig1-dev \
+            libfreetype6-dev \
+            libfribidi-dev \
+            libharfbuzz-dev \
+            libcurl4-openssl-dev \
+            libgit2-dev \
+            libicu-dev \
+            libjpeg-dev \
+            libpng-dev \
+            libtiff-dev \
+            libxml2-dev \
+            libssl-dev
+
+      - name: find polars rust source/cache, used by Makevars in check step
+        run: |
+          echo "RPOLARS_CARGO_CLEAN_DEPS=true" >> $GITHUB_ENV
+          echo "RPOLARS_RUST_SOURCE=${PWD}/src/rust" >> $GITHUB_ENV
+          echo "RPOLARS_RUST_SOURCE=${PWD}/src/rust"
+        shell: bash
+
+        ##### R stuff #####
+
+      - uses: r-lib/actions/setup-pandoc@v2
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: true
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          extra-packages: local::., any::pak
+      - name: Install required packages
+        run: |
+          pak::pak(c("etiennebacher/altdoc", "yaml", "magrittr", "cli",
+                     "tinkr", "stringr", "Genentech/rd2markdown@c4fb386"))
+        shell: Rscript {0}
+
+
+        ##### Build documentation and put it on "gh-pages" branch
+
+      - name: Build docs folder
+        run: |
+          altdoc::update_docs(custom_reference = "docs/make-docs.R")
+        shell: Rscript {0}
+
+        # https://www.mkdocs.org/user-guide/deploying-your-docs/
+      - name: Build site and deploy to GitHub pages
+        run: cd docs && mkdocs gh-deploy --force
```

Comment:
Note that we are not currently using the gh-pages branch of this repository.

https://github.com/pola-rs/r-polars/blob/2b04d3445dde103ff41099b91a7846eb4cc7f0af/.github/workflows/pkgdown.yaml#L107-L115